packages/core/src/external/carequality/ihe-gateway-v2/outbound/xca/process/dr-response.ts

```diff
@@ -93,9 +93,30 @@ async function processDocumentReference({
     const s3Utils = getS3UtilsInstance();
     const { mimeType, decodedBytes } = getMtomBytesAndMimeType(documentResponse, mtomResponse);
     const strippedDocUniqueId = stripUrnPrefix(documentResponse.DocumentUniqueId);
-    const metriportId = idMapping[strippedDocUniqueId];
+    let metriportId = idMapping[strippedDocUniqueId];
     if (!metriportId) {
-      throw new MetriportError("MetriportId not found for document");
+      const decodedBytesSizeInBytes = decodedBytes.length;
+      let closestSizeDiff = Infinity;
```

Comment:
shouldn't we move this inside the for loop?

if we send a DR with 5 doc refs, shouldn't we reasonably expect to get up to 5 docRefs in `documentResponse`? And if so, after we assign a match for one of the returned docs to an existing docRef, shouldn't we introduce logic that would make sure we don't pick that docRef again on the next object?